pkg/backup/backed_up_items_map.go

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright the Velero contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package backup
+
+import (
+	"fmt"
+	"sort"
+	"sync"
+)
+
+// backedUpItemsMap keeps track of the items already backed up for the current Velero Backup
+type backedUpItemsMap struct {
+	*sync.RWMutex
```

Comment:
Did we explore using `sync.Map` instead of `sync.RWMutex` ? 
https://pkg.go.dev/sync#Map

Reply:
Is one better than the other?

Reply:
Looks like maps fits well for this usecase if we [do not require type safety](https://pkg.go.dev/sync#Map:~:text=The%20Map%20type%20is%20optimized,separate%20Mutex%20or%20RWMutex.).

> The Map type is optimized for two common use cases: (1) when the entry for a given key is only ever written once but read many times, as in caches that only grow, or (2) when multiple goroutines read, write, and overwrite entries for disjoint sets of keys. In these two cases, use of a Map may significantly reduce lock contention compared to a Go map paired with a separate [Mutex](https://pkg.go.dev/sync#Mutex) or [RWMutex](https://pkg.go.dev/sync#RWMutex).

Reply:
@shubham-pampattiwar I had not explored that yet, but it's worth looking into. From what @kaovilai pasted above, this may well be a better choice.

Reply:
@shubham-pampattiwar upon further investigation, `sync.Map` doesn't appear to be a good choice for this use case. With `sync.Map`, individual updates and single key retrieval is synchronized, but using the `Range` func to iterate over the keys to produce an item list isn't locked for reading -- so this means the values can be written while iterating, leading to unpredictable results. I think I'll stick with the current implementation where the iteration-based funcs grab a Read lock for the full iteration.